core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -550,9 +550,16 @@ oppia.factory('explorationRightsService', [
         var that = this;
 
         requestParams.version = explorationData.data.version;
-        var explorationRightsUrl = (
-          '/createhandler/rights/' + explorationData.explorationId);
-        $http.put(explorationRightsUrl, requestParams).then(function(response) {
+        var requestUrl = '';
```

Comment:
put method in ExplorationPublishHandler handles publishing, unpublishing, publicizing and unpublicizing.

There are variables make_public/make_publicized (passed as request params) which decides what to do.

passing explicitly means changing url to /createhandler/publish/<make_public>?
If yes, then should another url like /createhander/publicize/<make_publicize> be introduced?

By public, do you mean (accessible from outside the EditorServices.js) or (functions accessible within the service and call them based on is_public/is_publicized) ?